examples/nvcc-test.py

```diff
@@ -72,55 +73,46 @@
-                [c.Value("T*", "inputPtr"),
-                 c.Value("int", "length"),
-                 c.Value("T*", "outputPtr")]))),
-    c.Block([c.Statement(global_index),
-           c.If("index == 0",
-              c.Statement("outputPtr[0] = inputPtr[0]"),
-              c.If("index < length",
-                 c.Statement(compute_diff),
-                 c.Statement("")))]),
+               CudaGlobal(c.FunctionDeclaration(c.Value("void", "diffKernel"),
+                          [c.Value("T*", "inputPtr"),
+                          c.Value("int", "length"),
+                          c.Value("T*", "outputPtr")]))),
+    c.Block([
+        c.Statement(global_index),
+        c.If("index == 0",
+        c.Statement("outputPtr[0] = inputPtr[0]"),
+        c.If("index < length",
+        c.Statement(compute_diff),
+        c.Statement("")))]),
 
     c.Template("typename T",
-                c.FunctionDeclaration(c.Value("CUdeviceptr", "difference"),
-                                          [c.Value("CUdeviceptr", "inputPtr"),
-                                           c.Value("int", "length")])),
+               c.FunctionDeclaration(c.Value("CUdeviceptr", "difference"),
+                                     [c.Value("CUdeviceptr", "inputPtr"),
+                                     c.Value("int", "length")])),
     c.Block([c.Statement(x) for x in launch])]
-
 cuda_mod.add_to_module(diff)
+
+# CudaModule.add_function also adds a declaration of this function to the
+# BoostPythonModule which is responsible for the host function.
+
 diff_instance = c.FunctionBody(
     c.FunctionDeclaration(c.Value("CUdeviceptr", "diffInstance"),
-                        [c.Value("CUdeviceptr", "inputPtr"),
-                         c.Value("int", "length")]),
+                          [c.Value("CUdeviceptr", "inputPtr"),
+                          c.Value("int", "length")]),
     c.Block([c.Statement("return difference<int>(inputPtr, length)")]))
-
-# CudaModule.add_function also adds a declaration of this
-# function to the BoostPythonModule which
-# is responsible for the host function.
 cuda_mod.add_function(diff_instance)
 
-import codepy.jit
-import codepy.toolchain
-
-
-gcc_toolchain = codepy.toolchain.guess_toolchain()
-nvcc_toolchain = codepy.toolchain.guess_nvcc_toolchain()
-
+gcc_toolchain = guess_toolchain()
+nvcc_toolchain = guess_nvcc_toolchain()
 module = cuda_mod.compile(gcc_toolchain, nvcc_toolchain, debug=True)
-import numpy as np
-import pycuda.autoinit
-import pycuda.driver
-import pycuda.gpuarray
-
 
-length = 25
-constant_value = 2
 # This is a strange way to create a GPUArray, but is meant to illustrate
 # how to construct a GPUArray if the GPU buffer it owns has been
 # created by something else
 
+length = 25
+constant_value = 2
+
 pointer = pycuda.driver.mem_alloc(length * 4)
 pycuda.driver.memset_d32(pointer, constant_value, length)
 a = pycuda.gpuarray.GPUArray((length,), np.int32, gpudata=pointer)
```

Comment:
This example doesn't work for me. It dies somewhere around here when calling `module.adjacentDifference(a)` with
```
TypeError: No registered converter was able to produce a C++ rvalue of type unsigned long long from this Python object of type DeviceAllocation
```

Not quite sure if that's some change in `pycuda` that wasn't updated here or what :(